### PR TITLE
Add __hash__ operator to Color class

### LINKED ISSUE #4454: `pydantic.color.Color` is not hashable in 1.10.0 release
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://pydantic-docs.helpmanual.io) and still think this is a bug
- [X] I am confident that the issue is with pydantic  (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

In pydantic `1.9.2` (and earlier), the Color class was Hashable. in release 1.10.0 it is not hashable that broke usage color as an argument to function decorated with `@lru_cache` from std lib. (https://github.com/napari/napari/issues/4995) 

A I do not see public method for mutate such object IO thin that it is deprecation. 

### Example Code

pydantic 1.9.2
```Python

In [1]: from pydantic.color import Color                                                     

In [2]: a = Color((0,0,0))                                                                   

In [3]: hash(a)                                                                              
Out[3]: 8771047785283
```

pydantic 1.10.0
```python
In [1]: from pydantic.color import Color                                                     

In [2]: a = Color((0,0,0))                                                                   

In [3]: hash(a)                                                                              
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-57b555d30865> in <module>
----> 1 hash(a)

TypeError: unhashable type: 'Color'
```
```


### Python, Pydantic & OS Version

```Text
pydantic 1.9.2 vs pydantic 1.10.0 tested on ubuntu and macos m1.
```

### Affected Components

- [X] [Compatibility between releases](https://pydantic-docs.helpmanual.io/changelog/)
- [ ] [Data validation/parsing](https://pydantic-docs.helpmanual.io/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://pydantic-docs.helpmanual.io/usage/exporting_models/) - `.dict()` and `.json()`
- [ ] [JSON Schema](https://pydantic-docs.helpmanual.io/usage/schema/)
- [ ] [Dataclasses](https://pydantic-docs.helpmanual.io/usage/dataclasses/)
- [ ] [Model Config](https://pydantic-docs.helpmanual.io/usage/model_config/)
- [x] [Field Types](https://pydantic-docs.helpmanual.io/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://pydantic-docs.helpmanual.io/usage/validation_decorator/)
- [ ] [Generic Models](https://pydantic-docs.helpmanual.io/usage/models/#generic-models)
- [X] [Other Model behaviour](https://pydantic-docs.helpmanual.io/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [ ] [Settings Management](https://pydantic-docs.helpmanual.io/usage/settings/)
- [ ] [Plugins](https://pydantic-docs.helpmanual.io/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

### PR DESCRIPTION
**Pull request by Czaki**
## Change Summary

Add `__hash__` method to `pydantic.color.Color` class.

## Related issue number

fix #4454

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/pydantic/pydantic/blob/main/changes/README.md) for details.
  You can [skip this check](https://github.com/pydantic/hooky#change-file-checks) if the change does not need a change file.)
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by Czaki** [comment]
please review

File: changes/4455-czaki.md [hunk 1]
```diff
@@ -0,0 +1 @@
+bugfix: Add `__hash__` operator to `pydancic.color.Color` class.
```
Comment (hramezani) [inline changes/4455-czaki.md:1]:
```suggestion
bugfix: Add `__hash__` method to `pydancic.color.Color` class.
```

**Comment by hramezani** [comment]
Thanks @Czaki for reporting the issue and preparing the fix :+1: 

I left a small suggestion.
Other than that LGTM

**Comment by samuelcolvin** [comment]
Thanks so much.

I haven't had any other bugs reported on v1.10 yet, so I'll make a patch release for just this.

**Pull request merged**